openedx_tagging/core/tagging/models/base.py

```diff
@@ -625,7 +465,19 @@ class Meta:
             models.Index(fields=["taxonomy", "object_id"]),
             models.Index(fields=["taxonomy", "_value"]),
         ]
-        unique_together = ("taxonomy", "_value", "object_id")
+        unique_together = [
+            ("object_id", "taxonomy", "tag_id"),
+            ("object_id", "taxonomy", "_value"),
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.pk:  # This is a new instance:
+            # Set _name and _value automatically on creation, if they weren't set:
+            if self.taxonomy and not self._name:
+                self._name = self.taxonomy.name
```

Comment:
nit (for symmetry with the stanza below, and to avoid fetching the taxonomy unless we need to):
```suggestion
            if not self._name and self.taxonomy:
                self._name = self.taxonomy.name
```